File: Nx3CS/bin/test/src/nx3/NScore.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class NScore : global::haxe.lang.HxObject {
		public    NScore(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    NScore(global::Array<object> nbars){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\NScore.hx"
				global::nx3.NScore.__hx_ctor_nx3_NScore(this, nbars);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_NScore(global::nx3.NScore __temp_me72, global::Array<object> nbars){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\NScore.hx"
				__temp_me72.nbars = nbars;
				{
					#line 20 "F:\\nx3\\src\\nx3\\NScore.hx"
					int _g = 0;
					#line 20 "F:\\nx3\\src\\nx3\\NScore.hx"
					while (( _g < nbars.length )){
						#line 20 "F:\\nx3\\src\\nx3\\NScore.hx"
						global::nx3.NBar bar = ((global::nx3.NBar) (nbars[_g]) );
						#line 20 "F:\\nx3\\src\\nx3\\NScore.hx"
						 ++ _g;
						#line 20 "F:\\nx3\\src\\nx3\\NScore.hx"
						bar.nscore = __temp_me72;
					}
					
				}
				
				#line 21 "F:\\nx3\\src\\nx3\\NScore.hx"
				__temp_me72.configuration = new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{}), new global::Array<object>(new object[]{}), new global::Array<int>(new int[]{}), new global::Array<double>(new double[]{}));
				global::haxe.lang.Runtime.setField(__temp_me72.configuration, "test", 1291438162, 123);
				global::haxe.lang.Runtime.setField(__temp_me72.configuration, "rtempo", 574624041, 80);
				global::haxe.lang.Runtime.setField(__temp_me72.configuration, "rlength", 597063384, 3);
				global::haxe.lang.Runtime.setField(__temp_me72.configuration, "rcountin", 365986882, 2);
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				return new global::nx3.NScore(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				return new global::nx3.NScore(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ));
			}
			#line default
		}
		
		
		public  global::Array<object> nbars;
		
		public  object configuration;
		
		public virtual   global::nx3.NBar getNBar(int idx){
			unchecked {
				#line 28 "F:\\nx3\\src\\nx3\\NScore.hx"
				if (( ( idx < 0 ) || ( idx > this.nbars.length ) )) {
					#line 28 "F:\\nx3\\src\\nx3\\NScore.hx"
					return default(global::nx3.NBar);
				}
				 else {
					#line 28 "F:\\nx3\\src\\nx3\\NScore.hx"
					return ((global::nx3.NBar) (this.nbars[idx]) );
				}
				
			}
			#line default
		}
		
		
		public virtual   object iterator(){
			unchecked {
				#line 30 "F:\\nx3\\src\\nx3\\NScore.hx"
				return ((object) (new global::_Array.ArrayIterator<object>(((global::Array<object>) (this.nbars) ))) );
			}
			#line default
		}
		
		
		public  int length;
		
		public virtual   int get_length(){
			unchecked {
				#line 32 "F:\\nx3\\src\\nx3\\NScore.hx"
				return this.nbars.length;
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						this.length = ((int) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return @value;
					}
					
					
					case 1037590710:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						this.configuration = ((object) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return @value;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						this.length = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return @value;
					}
					
					
					case 1037590710:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						this.configuration = ((object) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return @value;
					}
					
					
					case 388282030:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						this.nbars = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return @value;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				switch (hash){
					case 261031087:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("get_length") ), ((int) (261031087) ))) );
					}
					
					
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						if (handleProperties) {
							#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
							return this.get_length();
						}
						 else {
							#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
							return this.length;
						}
						
					}
					
					
					case 328878574:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("iterator") ), ((int) (328878574) ))) );
					}
					
					
					case 459968955:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNBar") ), ((int) (459968955) ))) );
					}
					
					
					case 1037590710:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return this.configuration;
					}
					
					
					case 388282030:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return this.nbars;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						if (handleProperties) {
							#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
							return ((double) (this.get_length()) );
						}
						 else {
							#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
							return ((double) (this.length) );
						}
						
					}
					
					
					case 1037590710:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return ((double) (global::haxe.lang.Runtime.toDouble(this.configuration)) );
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				switch (hash){
					case 261031087:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return this.get_length();
					}
					
					
					case 328878574:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return this.iterator();
					}
					
					
					case 459968955:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return this.getNBar(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ));
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				baseArr.push("length");
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				baseArr.push("configuration");
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				baseArr.push("nbars");
				#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
				{
					#line 10 "F:\\nx3\\src\\nx3\\NScore.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/NScore.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class NScore extends haxe.lang.HxObject
{
	public    NScore(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    NScore(haxe.root.Array<nx3.NBar> nbars)
	{
		nx3.NScore.__hx_ctor_nx3_NScore(this, nbars);
	}
	
	
	public static   void __hx_ctor_nx3_NScore(nx3.NScore __temp_me65, haxe.root.Array<nx3.NBar> nbars)
	{
		__temp_me65.nbars = nbars;
		{
			int _g = 0;
			while (( _g < nbars.length ))
			{
				nx3.NBar bar = nbars.__get(_g);
				 ++ _g;
				bar.nscore = __temp_me65;
			}
			
		}
		
		__temp_me65.configuration = new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{}));
		haxe.lang.Runtime.setField_f(__temp_me65.configuration, "test", ((double) (123) ));
		haxe.lang.Runtime.setField_f(__temp_me65.configuration, "rtempo", ((double) (80) ));
		haxe.lang.Runtime.setField_f(__temp_me65.configuration, "rlength", ((double) (3) ));
		haxe.lang.Runtime.setField_f(__temp_me65.configuration, "rcountin", ((double) (2) ));
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.NScore(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.NScore(((haxe.root.Array<nx3.NBar>) (arr.__get(0)) ));
	}
	
	
	public  haxe.root.Array<nx3.NBar> nbars;
	
	public  java.lang.Object configuration;
	
	public   nx3.NBar getNBar(int idx)
	{
		if (( ( idx < 0 ) || ( idx > this.nbars.length ) )) 
		{
			return null;
		}
		 else 
		{
			return this.nbars.__get(idx);
		}
		
	}
	
	
	public   java.lang.Object iterator()
	{
		return ((java.lang.Object) (new _Array.ArrayIterator<nx3.NBar>(((haxe.root.Array<nx3.NBar>) (this.nbars) ))) );
	}
	
	
	public  int length;
	
	public   int get_length()
	{
		return this.nbars.length;
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef425 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef425 = false;
						this.length = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1932752118:
				{
					if (field.equals("configuration")) 
					{
						__temp_executeDef425 = false;
						this.configuration = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef425) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef426 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef426 = false;
						this.length = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 104603694:
				{
					if (field.equals("nbars")) 
					{
						__temp_executeDef426 = false;
						this.nbars = ((haxe.root.Array<nx3.NBar>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1932752118:
				{
					if (field.equals("configuration")) 
					{
						__temp_executeDef426 = false;
						this.configuration = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef426) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef427 = true;
			switch (field.hashCode())
			{
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef427 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_length"))) );
					}
					
					break;
				}
				
				
				case 104603694:
				{
					if (field.equals("nbars")) 
					{
						__temp_executeDef427 = false;
						return this.nbars;
					}
					
					break;
				}
				
				
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef427 = false;
						if (handleProperties) 
						{
							return this.get_length();
						}
						 else 
						{
							return this.length;
						}
						
					}
					
					break;
				}
				
				
				case 1932752118:
				{
					if (field.equals("configuration")) 
					{
						__temp_executeDef427 = false;
						return this.configuration;
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef427 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("iterator"))) );
					}
					
					break;
				}
				
				
				case -75338437:
				{
					if (field.equals("getNBar")) 
					{
						__temp_executeDef427 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNBar"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef427) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef428 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef428 = false;
						if (handleProperties) 
						{
							return ((double) (this.get_length()) );
						}
						 else 
						{
							return ((double) (this.length) );
						}
						
					}
					
					break;
				}
				
				
				case 1932752118:
				{
					if (field.equals("configuration")) 
					{
						__temp_executeDef428 = false;
						return ((double) (haxe.lang.Runtime.toDouble(this.configuration)) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef428) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef429 = true;
			switch (field.hashCode())
			{
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef429 = false;
						return this.get_length();
					}
					
					break;
				}
				
				
				case -75338437:
				{
					if (field.equals("getNBar")) 
					{
						__temp_executeDef429 = false;
						return this.getNBar(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef429 = false;
						return this.iterator();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef429) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("length");
		baseArr.push("configuration");
		baseArr.push("nbars");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


